### DISCUSSION TITLE
How to use different model and gql types?

### DISCUSSION THREAD

**Discussion by bennyk8y** [dotansimha/graphql-code-generator#8683] [Q&A] [answered]
Hi!
Assuming this graphql schema:
```js
type GQLUser {
  fullName: String
}

type Query {
  getUser(args: Args): GQLUser
}
```

I have a case that there is a query resolver with this shape:
```js
Query: {
   getUser: (parents, args) => db.readUser(args); // <-- Error #1
  }
```

and assuming the db model looks like
```js
interface DBUser {
  name: string
}
```

And i want to use chained resolver this way:
```js
GQLUser: {
     fullName: (parent) => parent.name  // <-- Error #2
}
```

so the end result of resolvers looks like:
```js
import { Resolvers } from '__generated__';
const resolvers: Resolvers = {
  GQLUser: {
       fullName: (parent) => parent.name  // <-- Error #2
   },
  Query: {
     getUser: (parents, args) => db.readUser(args); // <-- Error #1
  }
};
```

As noted in the code, it raises 2 problems:
1. (error No' 1) The returned value of `getUser` is `DBUser` but in the schema (and in the generated types) is `GQLUser`. Codegen is not aware the there is a chained resolver that will transform `DBUser` result to the correct `GQLUser` type.
2. (error No' 2) The parent type in the resolver of `fullName` is `GQLUser` but i need it to be `DBUser` because this is what is being returned from `getUser`.

Tried to solve it by setting `allowParentTypeOverride: true` :
```js
import { UserResolvers, Resolvers } from '__generated__';

const userResolvers: UserResolvers<Context, DBUser> = {
   GQLUser: {
       fullName: (parent) => parent.name // <-- now it works
   }
};

const resolvers: Resolvers = {
  GQLUser: userResolvers, // <-- now this fails with Type 'UserResolvers<Context, DBUser>' is not assignable to type 'UserResolver<Context, GQLUser> | undefined'
   Query: {
     getUser: (parents, args) => db.readUser(args); // <-- still Error #1
   }
};
```

**Adding a mapper from GQLUser to DBUser wont help because i want to return GQLUser from `getUser` resolver** 

is there a way to solve these issues?

**Comment by bennyk8y** [comment, marked as answer]
After reading this: https://the-guild.dev/blog/better-type-safety-for-resolvers-with-graphql-codegen
i would like to add that the reason that i don't want to use mapper is because the `GQLUser` resolver will be omitted there wont be any TS error but clearly the return value of `getUser` would be wrong and gql will fail at runtime (because nothing will transform `DBUser` to `GQLUser`) and that's what im trying to avoid.

**Reply by gilgardosh** [reply]
I'm not sure I fully understand your issue, and the need of avoiding mappers.
I think [this video](https://www.youtube.com/watch?v=tHMaNmqPIC4&t=210s) examples it well
